docs/ops-recipe/harborDownOrUnhealthy.md

```diff
@@ -0,0 +1,313 @@
+
+FATAL: could not write lock file "postmaster.pid": No space left on device
+
+```
+
+We can adjust the memory by altering the fullstack-deployment manifest and redeploying it.
+
+```
+
+kind: Zlando/PostgreSQL
+
+spec:
+
+zlandoPostgreSql:
+
+operatorVersion: "1.8.2"
+
+storage: 2Gi <- Update this value to have more space available on the pod.
+
+replicas: 1
+
+image: registry.opensource.zalan.do/acid/spilo-14:2.1-p7
+
+```
+
+After increasing the storage ensure that you have backups configured for the `harbor-operator-user-values` configmap.
+
+```
+
+configAwsOrGcp:
+
+aws_region: us-east-1
+
+wal_s3_bucket: harbor-postgres-operator
+
+backup:
+
+enabled: true
+
+number_of_backups_to_retain: 5
+
+backup_schedule: "30 * * * *"
+
+bucket: "harbor-postgres-operator"
+
+aws:
+
+region: "us-east-1"
+
+restore:
+
+enabled: true
+
+```
+
+Having applied these two configurations we can delete the postgres and core pod in `harbor-cluster` namespace. If this doesn't solve the issue you may want to look into the state of the nodes and their available memory. As a last resort you can do a clean install of harbor with the backups and memory appropriately configured. A clean install with the right parameters, pointing to a valid backup, will restore projects and images. This means having the `restore` flag enabled. Configuration for things like harbor users may be lost.
+
+**Pod not in a running state**
+
+If you're trying to get a pod from a running to a ready state consider changing the thresholds for readiness probe. This isn't a viable long term fix, but could help with debugging. If pods are taking a long time to become ready it could be due to a high volume of objects in the etcd.
+
+```
+
+kubectl -n harbor-cluster edit pod harbor-$COMPONENT
+
+readinessProbe:
+  failureThreshold: 3
+  httpGet:
+    path: /api/v2.0/ping
+    port: http
+    scheme: HTTP
+  periodSeconds: 10
+  successThreshold: 1
+  timeoutSeconds: 1
+
+```
+
+From here you should increase the faliure threshold to something like 30 or increase the periodSeconds. This will give your pod more attempts to reach a ready state.
+
+During this time you can SSH onto the pod to retrieve more information about what is happening and why your pod may not be coming up.
+  
+
+It may be worth checking the kubelet logs on the node to see if there is any information about the healthiness of the pod.
+
+
+```
+
+kubectl -n harbor-cluster get pods -owide
+
+```
+
+Check what node the pod is associated with and SSH on to it. You can SSH onto the node using these [plugins](https://github.com/luksa/kubectl-plugins).
+
+```
+cat /var/logs/syslog | grep harbor
+
+```
+
+**Secrets**
+
+- Check whether there is a a secret in the harbor-cluster namespace containing the AWS authentication for the the s3 bucket used for backups.
+
+- Check if the secret is referenced in your pods definition. This can be configured in your fullstack for launching harbor under: storage -> s3.
+
+- Check that secrets exist for:
```

Comment:
What if they do not exist? Are they managed somewhere? What should they contain?